### PR TITLE
[TECHNICAL] Update Android Gradle Plugin version

### PR DESCRIPTION
**Pull request by JuancaG05**
## Related Issues
App: https://github.com/owncloud/android/issues/4478

- [x] Add changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
- [x] Add feature to Release Notes in `ReleaseNotesViewModel.kt` creating a new `ReleaseNote()` with String resources (if required)

_____

## QA

https://github.com/owncloud/android/pull/4507#issuecomment-2454737622

### CODE REVIEW

File: changelog/unreleased/4507 [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Enhancement: Upgraded AGP version to 8.7.2
+
+The Android Gradle Plugin version was upgraded to 8.7.2, together with Gradle version (updated to 8.9) and
```
Comment (joragua) [inline changelog/unreleased/4507:3]:
```suggestion
The Android Gradle Plugin version has been updated to 8.7.2, together with Gradle version (updated to 8.9) and
```
This should be in present perfect passive, right?

**Review by joragua — CHANGES_REQUESTED** [review]
A tiny comment to check here @JuancaG05! 🚀

File: changelog/unreleased/4507 (hunk 1 above)
Comment (JuancaG05) [reply changelog/unreleased/4507]: Done

**Review by joragua — APPROVED** [review]
LGTM! Ready for QA ✅

**Review by DeepDiver1975 — APPROVED** [review]
:+1:

**Comment by jesmrec** [comment]
### (1)

Using Android Studio Ladybug to generate a signed APK:

Lint is returning errors that don't allow to build:

```
Lint found fatal errors while assembling a release target.

Fix the issues identified by lint, or create a baseline to see only new errors.
To create a baseline, run `gradlew updateLintBaseline` after adding the following to the module's build.gradle file:

android {
    lint {
        baseline = file("lint-baseline.xml")
    }
}

For more details, see https://developer.android.com/studio/write/lint#snapshot

owncloudApp/src/main/java/com/owncloud/android/workers/DownloadFileWorker.kt:308: Error: Missing PendingIntent mutability flag [UnspecifiedImmutableFlag]
            PendingIntent.FLAG_ONE_SHOT
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~
owncloudApp/src/main/java/com/owncloud/android/workers/DownloadFileWorker.kt:329: Error: Missing PendingIntent mutability flag [UnspecifiedImmutableFlag]
            0
            ~

   Explanation for issues of type "UnspecifiedImmutableFlag":
   Apps targeting Android 12 and higher must specify either FLAG_IMMUTABLE or
   FLAG_MUTABLE when constructing a PendingIntent.

   FLAG_IMMUTABLE is available since target SDK 23, and is almost always the
   best choice. See
   https://developer.android.com/guide/components/intents-filters#CreateImmuta
   blePendingIntents for a list of common exceptions to this rule.

   https://developer.android.com/about/versions/12/behavior-changes-12#pending-intent-mutability
   https://goo.gle/UnspecifiedImmutableFlag

2 errors, 0 warnings
```

the problem seems to be more related with the android and SDK versions than the gradle one.. but `v4.4.1 ` was correctly built with the same method.

Debug builds (play button) work fine

**Comment by DeepDiver1975** [comment]
so apk build is not part of ci ... otherwise this would have been detected by ci - or am I missing anything? THX

**Comment by DeepDiver1975** [comment]
Is the difference signed vs not signed or debug vs release?

**Comment by jesmrec** [comment]
the difference in that case is signed vs not signed. It's posible to build release unsigned artifacts

**Comment by JuancaG05** [comment]
(1) should be fixed now @jesmrec

**Comment by jesmrec** [comment]
(1) is fixed

**Comment by jesmrec** [comment]
### (2)

The branding build system seems to need kind of update. It fails with the following trace:

```
* What went wrong:
A problem occurred evaluating project ':owncloudApp'.
> Failed to apply plugin 'com.android.internal.application'.
   > Android Gradle plugin requires Java 17 to run. You are currently using Java 11.
      Your current JDK is located in /usr/lib/jvm/java-11-openjdk-amd64
      You can try some of the following options:
       - changing the IDE settings.
       - changing the JAVA_HOME environment variable.
       - changing `org.gradle.java.home` in `gradle.properties`.
```

we'd need someone with access to the machine to update Java version to 17. Otherwise, it will not be posible to build branded artifacts anymore. Who is in change of this, @DeepDiver1975 ?

**Comment by jesmrec** [comment]
(2) will be fixed in the time till next release is done. Without a fix, the branding system will not work anymore. But, the app will go on with the new build setup. 

Let's move

**Pull request merged**